src/map/mob.c

```diff
@@ -312,6 +312,52 @@ static struct view_data *mob_get_viewdata(int class_)
 		return 0;
 	return &mob->db(class_)->vd;
 }
+
+/**
+ * Create unique view data associated to a spawned monster.
+ * @param md: Mob to adjust
+ */
+void mob_set_dynamic_viewdata(struct mob_data* md) {
+
+	nullpo_retv(md);
+
+	// If it is a valid monster and it has not already been created
+	if (!md->vd_changed) {
+		// Allocate a dynamic entry
+		struct view_data* vd = (struct view_data*)aMalloc(sizeof(struct view_data));
```

Comment:
if reload mob db will be memory leak and custom vd will be replaced to default vd

Reply:
also may be similar issue affected mob clones

Reply:
I addressed the remaining point about clones since it was still open